[16.0][ADD] product_simple_seasonality

kevinkhao opened the pull request:
![product](https://github.com/user-attachments/assets/65934a4e-fbff-4d0b-aaa4-3de7ee5dce09)

kevinkhao committed:
[16.0][ADD] product_simple_seasonality

bealdav commented on product_simple_seasonality/readme/CONFIGURE.md:
```diff
@@ -0,0 +1 @@
+Create seasonalities and assign them to products
```
```suggestion
Create seasonalities and assign them to products


Use case:

Season are recurring specific periods each year.
Apply product to season(s) allows to apply specific process to these products.

```

legalsylvain commented on product_simple_seasonality/models/seasonality.py:
```diff
@@ -0,0 +1,14 @@
+
+    name = fields.Char(required=True)
+    company_id = fields.Many2one(
+        "res.company", required=True, default=lambda self: self.env.company
```
```suggestion
        "res.company", default=lambda self: self.env.company
```

bealdav replied:
@kevinkhao why did you mark as resolved ?

kevinkhao replied:
dropped committed suggestion while rebasing. corrected now

legalsylvain reviewed (commented):
minor remarks inline. (I think seasonality could be global).

Otherwise, LGTM.

CC : @quentinDupont ?

bealdav commented on product_simple_seasonality/__manifest__.py:
```diff
@@ -0,0 +1,19 @@
+    "license": "AGPL-3",
+    "author": "Akretion,Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/product-attribute",
+    "depends": ["sale"],
```
```suggestion
    "depends": ["product"],
```

bealdav commented on product_simple_seasonality/views/product_template.xml:
```diff
@@ -0,0 +1,19 @@
+        <field name="inherit_id" ref="product.product_template_form_view" />
+        <field name="arch" type="xml">
+            <xpath expr="//group[@name='description']" position="inside">
+                <field name="seasonality_ids" />
```
Maybe we can have a widget with checkbox like route_ids field in product inventory tab ?

bealdav replied:
Maybe we can move towards optionnal_product_ids

quentinDupont replied:
I'd like more the many2manytags widget :)

kevinkhao replied:
after Upsell

bealdav commented on product_simple_seasonality/security/seasonality.xml:
```diff
@@ -0,0 +1,16 @@
+<odoo>
+
+    <record model="ir.model.access" id="campaign_seasonality_access_name">
+        <field name="name">campaign.seasonality sale user</field>
```
```suggestion
        <field name="name">Seasonality user</field>
```

bealdav commented on product_simple_seasonality/security/seasonality.xml:
```diff
@@ -0,0 +1,16 @@
+    <record model="ir.model.access" id="campaign_seasonality_access_name">
+        <field name="name">campaign.seasonality sale user</field>
+        <field name="model_id" ref="model_seasonality" />
+        <field name="group_id" ref="sales_team.group_sale_salesman" />
```
I think any employee can read, but a really small subset of user can CUD because we don't always create season. Which existing group (settings one ?)

bealdav commented on product_simple_seasonality/models/product_template.py:
```diff
@@ -0,0 +1,15 @@
+
+    seasonality_ids = fields.Many2many(
+        "seasonality",
+        string="Seasonality",
```
I don't remember if you don't set `string` param, default one is right.

bealdav commented on product_simple_seasonality/security/seasonality.xml:
```diff
@@ -0,0 +1,16 @@
+        <field name="perm_unlink" eval="1" />
+    </record>
+
+</odoo>
```
Below it seems that's not the right icon ?

bealdav commented on product_simple_seasonality/readme/CONTRIBUTORS.md:
```diff
@@ -0,0 +1,2 @@
+- Kevin Khao <kevin.khao@akretion.com>
+- David Beal <david.beal@akretion.com>
```
```suggestion
* `Akretion <https://akretion.com>`_:

  * Kevin Khao <kevin.khao@akretion.com>
  * David Beal <david.beal@akretion.com>

```

bealdav commented on product_simple_seasonality/readme/CONFIGURE.md:
```diff
@@ -0,0 +1,7 @@
+
+Season are recurring specific periods each year.
+
+Apply product to season(s) allows to apply specific process to these products.
```
Missing image(s) here

bealdav commented on product_simple_seasonality/__manifest__.py:
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "Product Simple Seasonality",
+    "summary": """
+        Product simple seasonality""",
```
```suggestion
        Product seasonality""",
```

PaulGoubert commented:
In term of analysis, add Seasonnality on Group By filter could be usefull. 
Moreover, have a link between Campain & seasonality could be also great, to sort different campaign by their seasonality, Don't you think? 
Otherwise, look good for me.

bealdav commented:
Campaign is not related to this module

PaulGoubert commented:
I hadn't noticed sale_seasonality.   Thank you

rousseldenis commented:
@kevinkhao @bealdav Wasn't it possible to improve product_seasonality in order to not have duplicate modules ?

eg.: removing some constraints like required date_start on configs and add the Many2many relation as you do it here

bealdav commented:
Hi @rousseldenis thanks for the review, sad to need to use `simple` in this module.
As I can see, previous product_seasonality merge `seasonality` "seasonal.config" and kind of `campaign like` "seasonal.config.line" (instance of seasonality for the last one) concepts.
We analysed this module.
Then it seems there are few chance other projects use this module even when they need seasonality.
There are a lot extra informations which can't match with our simple use case : see res_partner and res_company

If we split module, it'll complicate their project when they'll migrate. 

I think both use cases can exist and coexist : one for an advanced specific usage because they built it and their project required it and one for a very base usage on which everybody can define extra steps like use this seasonnality as parameter of supplychain.

Consider to check sale_seasonnality PR module as an extra light implementation of seasonnality in a very common way.

Maybe in future versions, the best ideas of both can be merged but we can't do that without them.

For now we need to go further and need to make accept a generic use of seasonnality with a basic common usage even if the name isn't perfect. Rome wasn't build in a day

my 2 cts

rousseldenis commented:
> Hi @rousseldenis thanks for the review, sad to need to use `simple` in this module. As I can see, previous product_seasonality merge `seasonality` "seasonal.config" and kind of `campaign like` "seasonal.config.line" (instance of seasonality for the last one) concepts. We analysed this module. Then it seems there are few chance other projects use this module even when they need seasonality. There are a lot extra informations which can't match with our simple use case : see res_partner and res_company
> 
> If we split module, it'll complicate their project when they'll migrate.
> 
> I think both use cases can exist and coexist : one for an advanced specific usage because they built it and their project required it and one for a very base usage on which everybody can define extra steps like use this seasonnality as parameter of supplychain.
> 
> Consider to check sale_seasonnality PR module as an extra light implementation of seasonnality in a very common way.
> 
> Maybe in future versions, the best ideas of both can be merged but we can't do that without them.
> 
> For now we need to go further and need to make accept a generic use of seasonnality with a basic common usage even if the name isn't perfect. Rome wasn't build in a day
> 
> my 2 cts

Indeed, I wanted to be sure something could not be done to merge/refactor/split.

Maybe something in ROADMAP for future migrators?

kevinkhao commented:
> > Hi @rousseldenis thanks for the review, sad to need to use `simple` in this module. As I can see, previous product_seasonality merge `seasonality` "seasonal.config" and kind of `campaign like` "seasonal.config.line" (instance of seasonality for the last one) concepts. We analysed this module. Then it seems there are few chance other projects use this module even when they need seasonality. There are a lot extra informations which can't match with our simple use case : see res_partner and res_company
> > If we split module, it'll complicate their project when they'll migrate.
> > I think both use cases can exist and coexist : one for an advanced specific usage because they built it and their project required it and one for a very base usage on which everybody can define extra steps like use this seasonnality as parameter of supplychain.
> > Consider to check sale_seasonnality PR module as an extra light implementation of seasonnality in a very common way.
> > Maybe in future versions, the best ideas of both can be merged but we can't do that without them.
> > For now we need to go further and need to make accept a generic use of seasonnality with a basic common usage even if the name isn't perfect. Rome wasn't build in a day
> > my 2 cts
> 
> Indeed, I wanted to be sure something could not be done to merge/refactor/split.
> 
> Maybe something in ROADMAP for future migrators?

Thanks for feedback. Added a note in ROADMAP.

bealdav commented:
Hi @rousseldenis all updates are done.

We can probably merge ?

Thanks


cc @legalsylvain

legalsylvain commented:
/ocabot merge nobump

(no review)
